identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp

```diff
@@ -105,6 +105,12 @@
         tenantDomain = srtenantDomain;
     }
 
+    if (username.contains(IdentityManagementEndpointConstants.TENANT_DOMAIN_SEPARATOR) && tenantDomain != null) {
```

Comment:
What are the scenarios that the tenant domain is null? If the tenant domain is carbon.super is it possible to tenant domain to be null.

Reply:
> What are the scenarios that the tenant domain is null? 

The null check was added to preserve the default behavior for the cases where the tenant domain is passed as null. For this particular case tenant domain is never null.

> If the tenant domain is carbon.super is it possible to tenant domain to be null.

In the tested JIT provisioning flow, when the tenant domain is carbon.super, the value will not be null.